### DISCUSSION TITLE
Print resolved config in DEBUG mode

### DISCUSSION THREAD

**Discussion by Wirone** [renovatebot/renovate#14701] [Suggest an Idea]
### What would you like Renovate to be able to do?

It would be helpful if Renovate could print resolved config (convert all options from `extends` to actual values) when `LOG_LEVEL=debug`.

I've spent some time debugging and reading why Renovate sets hourly limit for PRs to 2 when I did not set it explicitly. Then I found [this comment](https://github.com/renovatebot/renovate/issues/3833#issuecomment-497062709)... Yeah, we use `config:base` preset, but when I read job's logs it wasn't easy to find where `prHourlyLimit` comes from. It would be much easier if resolved config was in the output, so each explicitly/implicitly set option could be verified.

PS. I've searched through issues but did not find such request. Sorry in advance if it's duplicate.

### If you have any ideas on how this should be implemented, please tell us here.

Don't know codebase, so can't help with implementation, but in theory config IS resolved at some point, so it could be printed in the final form, so all configured options would be visible.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
Printing this would typically be too long and not worth it for the majority of users. You can add printConfig=true to your config instead.

**Reply by Wirone** [reply]
Thank you @rarkins 🍻  Somehow I did not manage to find this config option, but it's exactly what I needed. Just tested it in our Gitlab CI: triggered job with `RENOVATE_PRINT_CONFIG=true` and I see resolved config 🎉

**Reply by Wirone** [reply]
@rarkins additional question: if `RENOVATE_BASE_BRANCHES=some-branch` and `RENOVATE_USE_BASE_BRANCH_CONFIG=true` is set along with `RENOVATE_PRINT_CONFIG=true`, should it print resolved config for processed repository using `renovate.json` from `some-branch`? I wasn't able to test our config for MR with config changes, I always got current config (from main branch). So I've merged changes blindly, without knowing if everything is fine (there were our custom presets used from separate repo, with nested `extends`, validator was OK with them but I don't know if it works against actual, resolved result).

**Reply by rarkins** [reply]
I'm not certain, but I suspect that we don't validate or print the config in base branches. I think it's all done based on the default branch config.

**Reply by Wirone** [reply]
@rarkins should it be considered as bug? Or at least possible DX improvement? Because it's not good that it can't be verified before merging...

**Reply by rarkins** [reply]
Printing config per base branch would be a feature request, and please provide a public reproduction repo to run against with expected results.

For validating base branches, I'm not sure if it can/will be done. The validator today is not branch aware - it simply runs where you tell it. So the end user needs to make sure it's run in the right branch.
